apps/workshop/projects/SymbiosisFinance/modules/veSIS.ts

```diff
@@ -0,0 +1,83 @@
+import type { ModuleDefinitionInterface } from '@defiyield/sandbox';
+import { ADDRESS } from '../helpers/constants';
+import { Address, Pool } from '@defiyield/sandbox';
+import erc20Abi from '../../../../../packages/abis/erc20.abi.json';
+import veSISAbi from '../abis/veSIS.json';
+
+export const veSIS: ModuleDefinitionInterface = {
+  name: 'veSIS',
+  chain: 'ethereum',
+  type: 'staking',
+
+  /**
+   * Fetches the addresses of all involved tokens (supplied, rewards, borrowed, etc)
+   *
+   * @param context
+   * @returns Address[]
+   */
+  async preloadTokens(ctx): Promise<Address[]> {
+    return [ADDRESS.SIS];
+  },
+
+  /**
+   * Returns full pool list
+   *
+   * @param context
+   * @returns Pool[]
+   */
+  async fetchPools({ tokens, BigNumber, ethcall, ethcallProvider }): Promise<(Pool | void)[]> {
+    const [token] = tokens;
+    const sisContract = new ethcall.Contract(ADDRESS.SIS, erc20Abi);
+    const [locked] = await ethcallProvider.all<typeof BigNumber>([
+      sisContract.balanceOf(ADDRESS.veSIS),
+    ]);
+
+    const sisPrice = new BigNumber(token?.price || 0);
+    const sisDelimiter = new BigNumber(10).pow(token?.decimals);
+    const sisLocked = new BigNumber(locked.toString()).div(sisDelimiter);
+    const tvl = sisLocked.multipliedBy(sisPrice);
+
+    return [
+      {
+        id: 'veSIS',
+        supplied: [
+          {
+            token,
+            tvl: tvl.toNumber(),
+          },
```

Comment:
Would you like to show this pool in our explore opportunities? If yes, you need to attach the `apr.year` field here with APR

Reply:
Added